### PR TITLE
refactor: (De)Serialization of points using `GroupEncoding`

### PR DESCRIPTION
**Pull request by huitseeker**
## Context

In https://github.com/privacy-scaling-explorations/halo2curves/pull/30 and in https://github.com/privacy-scaling-explorations/halo2curves/pull/39 this repo implemented the `SerdeObject` that allows straight-line serialization and de-serialization of the internal point representation. This practice makes a lot of sense when (de)serialization is used to do local caching in zk-algorithms, and some sense when performing additions to e.g. a transcript.

However, it isn't compatible with classical point (de)serialization which more traditionally uses the compressed point representation, which is helpfully defined here as the implementation for the `zkcrypto/group::GroupEncoding` trait for both projective and affine point representations.

Precisely because the `SerdeObject` trait allows an alternate route to get the raw representation of point objects (and, IIUC, downstream repos use *these* code paths?), the default implementation of the `Serialize/Deserialize` traits could hence aim at compatibility and use the traditional compressed point representation for serde, which is what this PR suggests. 

In a nutshell, the `SerdeObject` code paths provide raw representation bytes, the `Serde` code paths provide the classical compressed point reresentations.

Because the implementation is limited to one single point of indirection (the serde annotation on `$name`, `$name_affine` becomes one macro call), we can also envision more optionality by e.g. opening a feature (e.g. "compressed_serde") allowing the user to pick and choose what serde would give them. 

## In detail
- Updated curve point (de)serialization logic from the internal representation to the representation offered by the implementation of the `GroupEncoding` trait.

### CODE REVIEW

**Review by CPerezz — COMMENTED** [review]
I like the direction this takes. 
In any case, will ask the review of other team members to see their opinions.

I like that `SerdeObject` (which we could rename TBH) serves as raw/unchecked encoding and that then, regular serde is used for canonically checked en/decoding. 
If we can also guarantee that with `GroupEncoding` we can both serialize and deserialize in compressed & uncompressed forms, that should be it.

**Review by han0110 — APPROVED** [review]
LGTM!

**Review by davidnevadoc — APPROVED** [review]
I agree with the rationale of the solution proposed in the description. LGTM!
